sql/catalyst/src/main/scala/org/apache/spark/sql/RowImplicits.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+object RowImplicits {
+
+  implicit class RowWithNamedFields[K, V](sqlRow: Row) {
```

Comment:
:+1: if so, I guess I should rename `getValue[T]` into an overloaded `getAs[T](fieldName: String)` to be consistent with an existing method `getAs`? (I remember overloading didn't work with implicits...)